### DISCUSSION TITLE
Suggestion for a better method name instead of firstUpdated in Reactive update cycle

### DISCUSSION THREAD

**Discussion by williamjacobszoon** [lit/lit#3036] [Ideas]
I know this sounds stupid but If the community can suggest and agree with me in renaming **firstUpdated** method in the Reactive update cycle to something better It would be great. ex: init

**Comment by drk-mtr** [comment]
My view: that's a poorer more confusing name, not a better one, because it will make people think of Angular's "ngOnInit" which is not equivalent to "firstUpdated".

Because you haven't given any rationale (and also because the current name is decent), I don't think the suggestion will get much traction.

**Reply by williamjacobszoon** [reply]
@drk-mtr method name **init** was just an example not a suggestion that I prefer myself. My concern was since **firstUpdated** only gets called once like a constructor naming with something else would be more suitable than calling **firstUpdated**.

**Reply by drk-mtr** [reply]
It gets called once when it's first updated, and the "first" makes it clear this only happens once.

Connected callback also gets called once - when it's connected to the DOM.

My interpretation is that the name makes this distinction clear.

**Reply by ynxtynxt** [reply]
My 2 cents, the arguments, firstUpdated(**changedProperties**), made clear that it is just an update**d**( changedProperties ) that came first. It's a kind of notification/callback so you should not suggest it with another role.

BTW, connnectedCallback can happen more than once while firstUpdated should happen only once, unless you set hasUpdated to false for any reason.
